cf-agent/files_editline.c

```diff
@@ -1470,34 +1479,76 @@ static int MatchPolicy(EvalContext *ctx, const char *camel, const char *haystack
                             while (isspace((int)*(sp + 1)))
                             {
                                 sp++;
+                                required_size++;
                             }
 
-                            strcat(spto, "\\s+");
-                            spto += 3;
+                            required_size += 3;
+                            strlcpy(toadd, "\\s+", sizeof(toadd));
                         }
                         else
                         {
-                            *spto++ = *sp;
+                            required_size++;
+                            toadd[0] = *sp;
+                            toadd[1] = '\0';
                         }
                     }
                     else
                     {
-                        *spto++ = *sp;
+                        required_size++;
+                        toadd[0] = *sp;
+                        toadd[1] = '\0';
+                    }
+
+                    if (required_size > work_size)
+                    {
+                        work_size = required_size + 12;
+                        work = xrealloc(work, work_size);
+                    }
+
+                    if (strlcat(spto, toadd, work_size) >= work_size - 1)
```

Comment:
It's a bit hard to follow the logic, but I think there may be a problem here: After this, the for loop ends and you increment ``spto`` by one. This will work if the character wasn't a space. However, if it was, and you added ``\s+`` here, ``spto`` will only have moved forward by one character, and the next write will overwrite the ``s``. Surely not the intention!